include/net/sntp.h

```diff
@@ -32,6 +32,12 @@ struct sntp_ctx {
 	u32_t expected_orig_ts;
 };
 
+/** SNTP timestamp, fractional seconds */
+struct sntp_tstamp {
```

Comment:
According to https://tools.ietf.org/html/rfc4330 chapter 3, the timestamp format is 64-bit value where the seconds part is in the first 32 bits and the fraction part in the last 32 bits.

Reply:
This has nothing to do with RFC4330. This is previous "u64_t epoch_time" augmented with fractional part. "sntp_" in the prefix there has the same meaning as in "sntp_ctx" and "sntp_request" (none of which has anything to do with RFC4330 and are part of Zephyr SNTP API, ditto for `struct sntp_tstamp`).

Reply:
Then the timestamp format should probably be done as described in that RFC. Otherwise it is hugely confusing if we are talking about different timestamps with different formats hers.

Reply:
> Then the timestamp format should probably be done as described in that RFC. Otherwise it is hugely confusing if we are talking about different timestamps with different formats hers.

So, the issue here is that this patch fixes the very exact issue, as linked in "Fixes" clause of the PR (#15596). And this issue is that the time (or timestamp) as returned by Zephyr SNTP API doesn't have precision. The time(stamp) is otherwise fully ok. So, what this patch does is change:

> was: `u64_t`, becomes `{ u64_t, u32_t }`

There's absolutely no intention (or need, IMHO) to do any further changes, definitely not in this patch.

If there's any confusion, then it seems to be about the name `struct sntp_tstamp`. Let me rename it to `struct sntp_time` and hope that fixes it (feel free to request it to be named `struct sntp_epoch_time` if you really like long names).

Reply:
Calling it `sntp_time` seems to be a good alternative here.